components/brave_new_tab_ui/components/search/config.ts

```diff
@@ -26,38 +22,57 @@ const getConfig = () => {
   return cache!
 }
 
-// Determines whether any search engines are enabled
-export const hasEnabledEngine = () => Object.values(getConfig()).find(c => c)
+export function useNtpWidgetDefaultSearchEngine() {
+  const [lastUsed, setLastUsed] = useNewTabPref('lastUsedNtpSearchEngine')
+  const lastLocalStorage = localStorage.getItem(LAST_SEARCH_ENGINE_KEY)
+  const config = getEnabledSearchEngines()
 
-export const maybeEnableDefaultEngine = () => {
-  // If no engines are enabled, enable the default one.
-  if (!hasEnabledEngine()) {
-    setEngineEnabled({ host: braveSearchHost }, true)
-  }
-}
+  const setLastNtpSearchEngine = React.useCallback((engine: SearchEngineInfo) => {
+    setLastUsed(engine.host);
+    localStorage.setItem(LAST_SEARCH_ENGINE_KEY, engine.host);
+  }, [setLastUsed])
 
-export const setEngineEnabled = (engine: { host: string }, enabled: boolean) => {
-  const config = getConfig()
-  config[engine.host] = enabled
+  const lastNtpSearchEngine = React.useCallback(() => {
```

Comment:
I think this function can be lifted out of the hook and just take `lastUsed` as an argument (as the other values are available directly from localStorage)

Reply:
I like it in the hook for the reason that I don't have to then also have `const [lastUsed, setLastUsed] = useNewTabPref('lastUsedNtpSearchEngine')` go everywhere with it. But if you feel strongly about this, I am open to changing it.